### DISCUSSION TITLE
Security verification with JWT on elsa-core dashboard and api

### DISCUSSION THREAD

**Discussion by fthmlymz** [elsa-workflows/elsa-core#2611] [Q&A]
Hi,

For elsa-core security, I want to validate with JWT on dashboard and swagger side. Is there a short way to do this?

Do not take any action without validation on both the dashboard(host ) file and the swagger. Is there a short way or example of doing this?

**Comment by sfmskywalker** [comment]
You can configure the dashboard to include a JWT with outgoing HTTP requests made to the Elsa server. Example: https://elsa-workflows.github.io/elsa-core/docs/next/extensibility/extensibility-designer-plugins#intercept-outgoing-http-requests

**Reply by fthmlymz** [reply]
Thank you very much for your interest.
But I want full jwt protection for designer and api.

**Reply by fthmlymz** [reply]
so I want to control all requests to be made via jwt.
https://localhost:2005/v1/signals/Approve/execute

and like all other api requests

**Reply by sfmskywalker** [reply]
Sure. You can secure your API controllers centrally form Startup using e.g.

```csharp
endpoints.MapControllers().RequireAuthorization();
```

Which is not ideal because this will require authorization on all of your API controllers.

We should probably decorate Elsa API controllers with a policy name, that can then be configured from Startup.
